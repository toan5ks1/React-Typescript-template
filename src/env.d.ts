// we will set any first, then change later
interface ImportMetaEnv {
    VITE_ENV: string;
    VITE_PJ_OWNER: string;
    VITE_STACK_DRIVER_KEY?: string;
    VITE_KEY_TOKEN?: string;
    VITE_KEY_PKG?: string;
    VITE_KEY_VERSION?: string;
    VITE_VALUE_PKG?: string;
    VITE_VALUE_VERSION?: string;
    VITE_PUBLIC_URL?: string;
    VITE_BUILD_VERSION?: string;
    VITE_NO_HASH?: string;
}
