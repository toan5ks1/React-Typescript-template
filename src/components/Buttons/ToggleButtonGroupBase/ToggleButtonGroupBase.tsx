import { arrayHasItem } from "src/common/utils/other";

import { ToggleButtonGroup, ToggleButtonGroupProps } from "@mui/material";
import { toggleButtonClasses } from "@mui/material/ToggleButton";
import { styled } from "@mui/material/styles";

import ToggleButtonBase, { ToggleButtonBaseProps } from "../ToggleButtonBase/ToggleButtonBase";

export interface ToggleButtonGroupBaseProps extends ToggleButtonGroupProps {
    options: ToggleButtonBaseProps[];
}

const StyledToggleButtonGroupBase = styled(ToggleButtonGroup)(({ theme }) => ({
    [`& .${toggleButtonClasses.selected}`]: {
        borderColor: `${theme.palette.primary.border} !important`,
    },
}));

const ToggleButtonGroupBase = (props: ToggleButtonGroupBaseProps) => {
    const { options, ...rest } = props;

    return (
        <StyledToggleButtonGroupBase
            exclusive
            size="small"
            data-testid="ToggleButtonGroupBase"
            {...rest}
        >
            {arrayHasItem(options) &&
                options.map(({ children, value, ...rest }, index) => {
                    return (
                        <ToggleButtonBase key={index} value={value} {...rest}>
                            {children}
                        </ToggleButtonBase>
                    );
                })}
        </StyledToggleButtonGroupBase>
    );
};

export default ToggleButtonGroupBase;
