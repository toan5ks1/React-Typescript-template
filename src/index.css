body {
    margin: 0;
    font-family: "Roboto", sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    overflow-x: hidden;
}

code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace;
}

body,
html,
#root {
    margin: 0;
    padding: 0;
    width: 100%;
    min-height: 100vh;
    background-color: #fff;
}

#root > div:first-child {
    margin: 0;
    padding: 0;
    width: 100%;
    min-height: 100vh;
    box-sizing: border-box;
}

.list-page > div > div:last-child {
    overflow: auto;
}

table td {
    max-width: 300px;
    overflow-wrap: break-word;
}

/* Override css background input autocomplete - Designer wants it */
input:-webkit-autofill,
input:-webkit-autofill:hover,
input:-webkit-autofill:focus,
textarea:-webkit-autofill,
textarea:-webkit-autofill:hover,
textarea:-webkit-autofill:focus,
select:-webkit-autofill,
select:-webkit-autofill:hover,
select:-webkit-autofill:focus {
    -webkit-box-shadow: 0 0 0px 1000px #fff inset;
    transition: background-color 5000s ease-in-out 0s;
}

/* styled not affectted, so we use this, check src/components/BrightcoveVideo/BrightcoveVideo.tsx */
.bc-player-default_default {
    width: 100%;
    height: 100%;
}
