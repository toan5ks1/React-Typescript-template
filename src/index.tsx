import "src/styles/mui-classname-setup";

import { StrictMode } from "react";

import { enableMapSet } from "immer";
import ReactDOM from "react-dom";
import enableSidebarMethod from "src/squads/architecture/internals/sidebar";
import enableManaEventEmitter from "src/squads/root/src/internals/mana-event-emitter";

import App from "./app";
import AppProvider from "./app/AppProvider";
import "./index.css";
import { enableReport } from "./internals/stackdriver";
import { enableLogger } from "./internals/warner";

enableMapSet();
enableReport();
enableLogger();

enableSidebarMethod();
enableManaEventEmitter();

ReactDOM.render(
    <StrictMode>
        <AppProvider>
            <App />
        </AppProvider>
    </StrictMode>,
    document.getElementById("root")
);
