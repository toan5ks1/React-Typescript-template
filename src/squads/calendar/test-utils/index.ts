export { default as TestThemeProvider } from "./TestThemeProvider";
export { default as TestApp } from "./TestApp";
export { default as AppProvider } from "./AppProvider";
