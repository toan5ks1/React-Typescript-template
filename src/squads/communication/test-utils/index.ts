export { default as TestThemeProvider } from "./TestThemeProvider";
export { default as TestApp } from "./TestApp";
