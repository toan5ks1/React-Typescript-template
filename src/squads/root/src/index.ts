import { createBrowserHistory } from "history";
import { start, registerApplication } from "single-spa";

const routeHistory = createBrowserHistory();

//
// registerApplication(
//     "syllabus",
//     () => {
//         // @ts-ignore
//         // eslint-disable-next-line import/no-unresolved
//         return import("@manabie-com/mfe/syllabus");
//     },
//     (location) => location.pathname.startsWith("/syllabus"),
//     {
//         rootElement: "#root1",
//     }
// );
//
registerApplication(
    "user",
    () => {
        // @ts-ignore
        // eslint-disable-next-line import/no-unresolved
        return import("@manabie-com/mfe/user");
    },
    (location) => location.pathname.startsWith("/user") || location.pathname === "/",
    {
        rootElement: "#root1",
        basename: "/user",
        routeHistory: routeHistory,
    }
);
//
// registerApplication(
//     "adobo",
//     () => {
//         // @ts-ignore
//         // eslint-disable-next-line import/no-unresolved
//         return import("@manabie-com/mfe/adobo");
//     },
//     (location) => location.pathname.startsWith("/adobo"),
//     {
//         rootElement: "#root1",
//     }
// );

start({
    urlRerouteOnly: true,
});
