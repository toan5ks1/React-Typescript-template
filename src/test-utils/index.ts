export { default as TestThemeProvider } from "./TestThemeProvider";
export { default as AppProvider } from "./AppProvider";
export { default as TestApp } from "./TestApp";
export * from "./DOM";
