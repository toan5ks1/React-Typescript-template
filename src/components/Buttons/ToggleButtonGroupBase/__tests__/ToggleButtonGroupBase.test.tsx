import { TestThemeProvider } from "src/test-utils";

import ToggleButtonGroupBase, { ToggleButtonGroupBaseProps } from "../ToggleButtonGroupBase";

import { fireEvent, render, RenderResult, screen } from "@testing-library/react";

describe("<ToggleButtonGroupBase />", () => {
    let wrapper: RenderResult;

    const props: ToggleButtonGroupBaseProps = {
        options: [
            { value: "test1", children: "Test 1" },
            { value: "test2", children: "Test 2" },
        ],
        value: "test1",
        onChange: jest.fn(),
    };

    beforeEach(() => {
        wrapper = render(
            // Add Provider to make the test matches with our theme color
            <TestThemeProvider>
                <ToggleButtonGroupBase {...props} />
            </TestThemeProvider>
        );
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });

    it("should exist children", () => {
        expect(screen.getByText("Test 1")).toBeInTheDocument();
        expect(screen.getByText("Test 2")).toBeInTheDocument();
    });

    it("should match color style", () => {
        expect(wrapper.getAllByTestId("ToggleButtonBase")[0]).toHaveStyle(
            "color: rgb(33, 150, 243)"
        ); // active toggleButton
        expect(wrapper.getAllByTestId("ToggleButtonBase")[1]).toHaveStyle("color: #757575");
    });

    it("should call onChange", () => {
        fireEvent.click(wrapper.getAllByTestId("ToggleButtonBase")[1]);
        expect(props.onChange).toBeCalled();
    });
});
