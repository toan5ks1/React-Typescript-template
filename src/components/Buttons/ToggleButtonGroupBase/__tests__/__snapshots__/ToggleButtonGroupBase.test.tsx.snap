// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButtonGroupBase /> should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 4px;
}

.emotion-0 .MuiToggleButtonGroup-grouped:not(:first-of-type) {
  margin-left: -1px;
  border-left: 1px solid transparent;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.emotion-0 .MuiToggleButtonGroup-grouped:not(:last-of-type) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-0 .MuiToggleButtonGroup-grouped.Mui-selected+.MuiToggleButtonGroup-grouped.Mui-selected {
  border-left: 0;
  margin-left: 0;
}

.emotion-0 .Mui-selected {
  border-color: #90CAF9!important;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-size: 0.8125rem;
  font-weight: 500;
  letter-spacing: 0.4px;
  text-transform: capitalize;
  font-family: Roboto,sans-serif;
  line-height: 1.75;
  border-radius: 4px;
  padding: 7px;
  border: 1px solid #E0E0E0;
  color: rgba(0, 0, 0, 0.54);
}

.emotion-1::-moz-focus-inner {
  border-style: none;
}

.emotion-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-1.Mui-disabled {
  color: #BDBDBD;
  border: 1px solid #E0E0E0;
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(33, 33, 33, 0.04);
}

@media (hover: none) {
  .emotion-1:hover {
    background-color: transparent;
  }
}

.emotion-1.Mui-selected {
  color: #212121;
  background-color: rgba(33, 33, 33, 0.08);
}

.emotion-1.Mui-selected:hover {
  background-color: rgba(33, 33, 33, 0.12);
}

@media (hover: none) {
  .emotion-1.Mui-selected:hover {
    background-color: rgba(33, 33, 33, 0.08);
  }
}

.emotion-1.ToggleButtonBase-root {
  color: #757575;
  padding: 6px 1rem;
}

.emotion-1.ToggleButtonBase-selected {
  color: #2196F3;
  background-color: #EDF7FE;
  border-color: #90CAF9;
}

.emotion-2 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div
    class="MuiToggleButtonGroup-root emotion-0"
    data-testid="ToggleButtonGroupBase"
    role="group"
  >
    <button
      aria-pressed="true"
      class="MuiButtonBase-root ToggleButtonBase-root MuiToggleButton-root ToggleButtonBase-selected Mui-selected MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-1"
      data-testid="ToggleButtonBase"
      tabindex="0"
      type="button"
      value="test1"
    >
      Test 1
      <span
        class="MuiTouchRipple-root emotion-2"
      />
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root ToggleButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal emotion-1"
      data-testid="ToggleButtonBase"
      tabindex="0"
      type="button"
      value="test2"
    >
      Test 2
      <span
        class="MuiTouchRipple-root emotion-2"
      />
    </button>
  </div>
</div>
`;
